.github/workflows/docker_publish.yml

```diff
@@ -30,6 +30,7 @@ jobs:
           5.5.1,
         ]
         geant4_version : [
+          off,
```

Comment:
Not sure it's worth have an extra image without GEANT?

Reply:
While updating [action.yml](https://github.com/svalinn/DAGMC/blob/develop/.github/actions/upstream-test/action.yml#L11), I have noticed we are not using Geant4 by default. Do you like to change the default?

Reply:
The treatment of GEANT and Double Down are a little different from each other.

1) Testing with or without GEANT only demonstrates whether DAGMC will integrate successfully with GEANT as a transport code.

2) Testing with or without Double Down modifies the underlying fundamental behavior of DAGMC with all transport codes.

A few conclusions of this:

A) It makes sense to have fewer images, i.e. none that are build without GEANT and without DD because we can test our build without them even if they are installed.

B) It makes sense to do the advisory testing with the minimal number of options turned on, ie. without GEANT

Reply:
I've made changes to streamline the build process and reduce the number of arguments as much as possible. But it seems, I've ended up creating extra Docker images. 🙂

I think we have two options:

1. Add the extra arguments again and remove conditions from the Dockerfile.
2. Consider setting GEANT4 and DD always on in `action.yml`.

Reply:
Let me know, which step to proceed. Or, any other suggestions?